js/main.ts

```diff
@@ -19,6 +20,10 @@ import * as deno from "./deno";
 import libDts from "gen/cli/lib/lib.deno_runtime.d.ts!string";
 
 export default function denoMain(name?: string): void {
+  // Deno.core could ONLY be safely frozen here (not in globals.ts)
+  // since shared_queue.js will modify core properties.
+  Object.freeze(window.Deno.core);
```

Comment:
Maybe `os.start()` is a better place for this? I think this won't get called for workers.

Reply:
Actually `denoMain()` is still called in user-created workers:

https://github.com/denoland/deno/blob/0796a8f2f75005df95ef6115a4bdf6dd66e58dc3/cli/ops.rs#L1878-L1881

but yeah putting it in `os.start()` sound better since we also call `core.setAsyncHandler(handleAsyncMsgFromRust);` for `core`-related initialization.